src/cpu/ppc64/ppc64_gemm_s8x8s32.hpp

```diff
@@ -20,2108 +20,1255 @@
 namespace dnnl {
 namespace impl {
 
-typedef int int_A1 __attribute__((aligned(1)));
-#define memcpy_4(_d, _s) *((int_A1 *)(_d)) = *((int_A1 *)(_s));
-
-#define memcpy_14(_d, _s) \
-    { \
-        int_A1 *_di, *_si; \
-        _di = (int *)(_d); \
-        _si = (int *)(_s); \
-        *_di++ = *_si++; \
-        *_di++ = *_si++; \
-        *_di++ = *_si++; \
-        *((short *)(_di)) = *((short *)(_si)); \
-    }
+static inline void memcpy_4(void * __restrict d, const void * __restrict s) {
+        int *di, *si;
+        di = (int *)(d);
+        si = (int *)(s);
+        di[0] = si[0];
+}
 
-#define memcpy_16(_d, _s) \
-    { \
-        int_A1 *_di, *_si; \
-        _di = (int *)(_d); \
-        _si = (int *)(_s); \
-        _di[0] = _si[0]; \
-        _di[1] = _si[1]; \
-        _di[2] = _si[2]; \
-        _di[3] = _si[3]; \
-    }
+static inline void memcpy_14(void * __restrict d, const void * __restrict s) {
+        int *di, *si;
+        di = (int *)(d);
+        si = (int *)(s);
+        di[0] = si[0];
+        di[1] = si[1];
+        di[2] = si[2];
+        *(short *) &di[3] = *(short *) &si[3];
+}
 
-#define memcpy_32(_d, _s) \
-    { \
-        int_A1 *_di, *_si; \
-        _di = (int *)(_d); \
-        _si = (int *)(_s); \
-        _di[0] = _si[0]; \
-        _di[1] = _si[1]; \
-        _di[2] = _si[2]; \
-        _di[3] = _si[3]; \
-        _di[4] = _si[4]; \
-        _di[5] = _si[5]; \
-        _di[6] = _si[6]; \
-        _di[7] = _si[7]; \
-    }
+static inline void memcpy_16(void * __restrict d, const void * __restrict s) {
+        int *di, *si;
+        di = (int *)(d);
```

Comment:
(nit) use int32_t to specify the type size?
Also, compilers have optimized/vectorize implementations for memcpy. [Here ](https://godbolt.org/z/TeMPrvo1f) is an example on x64, where a call to memcpy with fixed size are just swapped with a single instruction.

Alternatively, I think 4 calls to memcpy_4 should produce the same code for most compilers. [Here ](https://godbolt.org/z/d9WEbTE1P) is an example.